tools/tests/py2fgen/wrappers/test_icon4py_export.py

```diff
@@ -0,0 +1,57 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+import cffi
+from gt4py import next as gtx
+from gt4py.next.type_system import type_specifications as ts
+
+from icon4py.tools import py2fgen
+from icon4py.tools.py2fgen.wrappers import icon4py_export
+
+
+export_with_mapping_hook = py2fgen.export(
+    annotation_mapping_hook=icon4py_export.field_annotation_mapping_hook,
+    param_descriptors={
+        "a": py2fgen.ArrayParamDescriptor(
+            rank=1, dtype=ts.ScalarKind.INT32, device=py2fgen.DeviceType.HOST, is_optional=False
+        ),
+        "b": py2fgen.ScalarParamDescriptor(dtype=ts.ScalarKind.INT32),
+    },
+)
+
+SomeDim = gtx.Dimension("SomeDim")
+
+
+def make_array_info(
+    ptr: "cffi.FFI.CData",  # TODO don't `from __future__ import annotations` otherwise the gt4py annotation will be a string
```

Comment:
I guess this TODO is related to the previous TODO about switching from `inspect.signature()` to `typing.get_type_hints()`. I think you should either address both issues or add a mention here to the other TODO (or the other way around)